src/components/HomePage.jsx

```diff
@@ -0,0 +1,26 @@
+import { useEffect } from "react";
+import { Text, View } from "react-native"
+import { useNavigate } from '../Router'
+import { styles } from '../styling/styles'
+
+const HomePage = (props) => {
+  const navigate = useNavigate();
+  useEffect(() => {
+    if (!props.user) {
+      navigate('/login')
+    }
+  }, [])
+
+  return !props.user
+    ? navigate('/login') 
```

Comment:
maybe not return the result of `navigate` which [seems to be](https://reactrouter.com/en/main/hooks/use-navigate#usenavigate) `void`?